pyproject.toml

```diff
@@ -14,20 +14,24 @@ include = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.7.1"
-numpy = "^1.20"
-setuptools = "^51.1.2"
-pytest-fail-slow = "^0.2.0"
+python = ">=3.7,<4.0"
+numpy = [
+    {version = ">=1.20", python = ">=3.7,<3.10"},
+    {version = ">=1.23", python = ">=3.10"}
+]
+setuptools = ">=51.1.2"
 
 [tool.poetry.dev-dependencies]
```

Comment:
not sure why these need to be changed?

Reply:
I changed setuptools (and other things) to be >= rather than ~ as I feel the upper bound constraints are too restrictive (see https://iscinumpy.dev/post/bound-version-constraints/). For example, the latest version of setuptools is 67.1.0 and the exact-cover package works absolutely fine with that. But the current pyproject.toml insists one downgrades to 51.3.3 or earlier as it is written ~51.1.2

The higher numpy versioning is for easier install on more recent pythons. There are no binary wheels for numpy 1.20 for python 3.10 and up. When poetry does its dependency resolving it fetches an old version of numpy that it has to compile from source (which caused difficulty on the CI windows builds). My change here was just to force the newer python to use newer numpy where binary wheels are available.